src/transposition.cpp

```diff
@@ -1090,6 +1169,17 @@ std::string Transposer::CircleOfFifthsToIntervalName(int fifths)
     return GetIntervalName(intervalClass);
 }
 
+//////////////////////////////
+//
+// Transposer::CircleOfFifthsToPitch -- Assuming the mode is major, guess the pitch
+// from the circle of fifths position.
+//
+TransPitch Transposer::CircleOfFifthsToPitch(int fifths)
+{
+    int intervalClass = CircleOfFifthsToIntervalClass(fifths);
+    return IntegerToPitch(GetMaxAccid() + 1 + intervalClass);
```

Comment:
@craigsapp There's probably a better way of doing this. (I think this is needed because `0` is equivalent to Cbbb instead of C).

Reply:
You are correct: C is not the zero point in the octave cycle, but rather the flattest C possible in the given base.  This is to allow easier calculating of the octave as compared to setting C-natural to be the zero point (Cb4 is enharmonically equivalent to B3, Cbb4 is enharmonically equivalent to Bb3).  So as you notice, the interval class and the pitch class integers are shifted from each other (P1 = 0, but C = max_accid).

However, I do not think the `+ 1` is needed: In Base-40: Cbb=0, Cb=1, C=2.  And in base-40 the maximum accidental is 2, so the C pitch class in any base is the maximum accidental count, not the maximum accidental plus one.

Reply:
Also, the assumption about the mode being major is not necessary (or used in your code): the transposition is calculated from the interval class between the pitch classes of the tonics of the two keys.   The tonic is independent of the mode (which instead describes the pitch classes of other notes in the scale).  In other words, in C major, there is a C-natural, and in C minor, there is a C-natural, and in C dorian, there is a C-natural, and in C create-your-own-mode, there is a C natural.  Only for non-tonic notes can the accidental state of the notes change: in C major, there is an E-natural, but in C minor, there is an E-flat.

Reply:
When I don't use the `+ 1`, the pitch comes out flat.

From printing out some debug output, it seems like `IntegerToPitch` (or `GetMaxAccid`) is not behaving as you explained: `GetMaxAccid()` is returning 2, but `IntegerToPitch(0)` is returning `Cbbb0` (triple-flat)

```c++
    Transposer transposer;
    bool setTransp = transposer.SetTransposition(this->m_options->m_transpose.GetValue());
    std::cerr << "transposer.GetMaxAccid(): ";
    std::cerr << transposer.GetMaxAccid() << "\n";
    std::cerr << "transposer.IntegerToPitch(0): ";
    std::cerr << transposer.IntegerToPitch(0) << "\n";
    std::cerr << "transposer.IntegerToPitch(1): ";
    std::cerr << transposer.IntegerToPitch(1) << "\n";
    std::cerr << "transposer.IntegerToPitch(2): ";
    std::cerr << transposer.IntegerToPitch(2) << "\n";
    std::cerr << "transposer.IntegerToPitch(3): ";
    std::cerr << transposer.IntegerToPitch(3) << "\n";
```

```
transposer.GetMaxAccid(): 2
transposer.IntegerToPitch(0): Cbbb0
transposer.IntegerToPitch(1): Cbb0
transposer.IntegerToPitch(2): Cb0
transposer.IntegerToPitch(3): C0
```

Reply:
Hmmm, I am probably subtracting one incorrectly somewhere in the Transpose class: I will check it out.

Reply:
It looks like it needs to be shifted back one, since B## is not accessible in base-40 (The off-by-one issue also applies to base-600)

```
transposer.IntegerToPitch(0): Cbbb0
transposer.IntegerToPitch(1): Cbb0
transposer.IntegerToPitch(2): Cb0
transposer.IntegerToPitch(3): C0
transposer.IntegerToPitch(4): C#0
transposer.IntegerToPitch(5): C##0
transposer.IntegerToPitch(6): C###0
transposer.IntegerToPitch(7): Dbb0
transposer.IntegerToPitch(8): Db0
transposer.IntegerToPitch(9): D0
transposer.IntegerToPitch(10): D#0
transposer.IntegerToPitch(11): D##0
transposer.IntegerToPitch(12): D###0
transposer.IntegerToPitch(13): Ebb0
transposer.IntegerToPitch(14): Eb0
transposer.IntegerToPitch(15): E0
transposer.IntegerToPitch(16): E#0
transposer.IntegerToPitch(17): E##0
transposer.IntegerToPitch(18): Fbb0
transposer.IntegerToPitch(19): Fb0
transposer.IntegerToPitch(20): F0
transposer.IntegerToPitch(21): F#0
transposer.IntegerToPitch(22): F##0
transposer.IntegerToPitch(23): Gbbb0
transposer.IntegerToPitch(24): Gbb0
transposer.IntegerToPitch(25): Gb0
transposer.IntegerToPitch(26): G0
transposer.IntegerToPitch(27): G#0
transposer.IntegerToPitch(28): G##0
transposer.IntegerToPitch(29): Abbb0
transposer.IntegerToPitch(30): Abb0
transposer.IntegerToPitch(31): Ab0
transposer.IntegerToPitch(32): A0
transposer.IntegerToPitch(33): A#0
transposer.IntegerToPitch(34): A##0
transposer.IntegerToPitch(35): Bbbb0
transposer.IntegerToPitch(36): Bbb0
transposer.IntegerToPitch(37): Bb0
transposer.IntegerToPitch(38): B0
transposer.IntegerToPitch(39): B#0
transposer.IntegerToPitch(40): Cbbb1
transposer.IntegerToPitch(41): Cbb1
transposer.IntegerToPitch(42): Cb1
```

Reply:
The problem should be fixed with commit https://github.com/rism-ch/verovio/pull/1242/commits/2a621aba9148db9236e791cd1e508e931e6a1504

When I was calculating the interval classes of the diatonic pitches, I started with C = maxAccid + 1, where it should have been C = maxAccid.  In other words, I did the same thing that I was complaining about in your code (so you had to add one to fix the problem).

I will fix the +1 that I was complaining about in another commit.

Reply:
Got it in 1d2c4cc4809dbd2d2497c06770be32294f1c3786

Reply:
But you beat me to it.

Reply:
I recommend changing the name of `Transposer::IntegerToPitch()` to `Transposer::PitchClassToTransPitch()`.  This will help with possible confusion between interval classes and pitch classes: both of these are integers.  The name `IntegerToPitch` might imply to someone that both interval class or pitch class integers can be given to the function, but the two are not interchangeable (particularly due to the offset added to pitch classes).

Reply:
Actually changed `IntegerToPitch` to be `IntegerPitchToTransPitch`, since it is not a pitch class but rather a pitch (includes octave).